File: open-source-analysis/Dagger2AndroidInjection-v2.19-Sample/base/src/main/java/com/ztiany/base/AppContext.java
package com.ztiany.base;

import android.app.Application;
import android.content.Context;

import dagger.android.HasActivityInjector;

/**
 * @author Ztiany
 *         Email: ztiany3@gmail.com
 *         Date : 2017-05-22 11:02
 */
public abstract class AppContext extends Application implements HasActivityInjector{

    private static Context sContext;

    @Override
    public void onCreate() {
        super.onCreate();
        CrashHandler.register(this);
        sContext = this;
    }

    public static Context getContext() {
        return sContext;
    }

}

File: architecture-components/app/src/main/java/com/ztiany/androidx/AppContext.kt
package com.ztiany.androidx

import android.app.Application
import android.content.Context
import androidx.appcompat.app.AppCompatDelegate
import androidx.multidex.MultiDexApplication
import dagger.hilt.android.HiltAndroidApp
import kotlin.properties.Delegates

/**
 *@author Ztiany
 */
@HiltAndroidApp
class AppContext : MultiDexApplication() {

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        application = this
    }

    override fun onCreate() {
        super.onCreate()
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
    }

    companion object {

        private var application by Delegates.notNull<Application>()

        fun getInstance(): Application {
            return application
        }

    }

}